### PR TITLE
feat: order build for apm transaction

### PR DESCRIPTION
**Pull request by pietro-tota**
Depends on https://github.com/pagopa/pagopa-ecommerce-commons/pull/164
eCommerce local NPG mock update PR: https://github.com/pagopa/pagopa-ecommerce-local/pull/118

#### List of Changes
Add logic to handle APM NPG authorization requests for payments performed with wallet.
For APM wallet payments the a `POST order/builds` will be performed to NPG with amount valued with transaction total amount expecting a response with state `REDIRECT_TO_EXTERNAL_DOMAINS` and url pointing to the redirection url to which navigate in order to perform transaction payment.

#### Motivation and Context
Those modifications are needed in order to properly handle authorization requests with onboarded APM wallet

#### How Has This Been Tested?
Junit tests + local tests with eCommerce local and mocked NPG responses

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -485,6 +485,8 @@
                     <connectionUrl>scm:git:https://github.com/pagopa/pagopa-ecommerce-commons.git</connectionUrl>
                     <scmVersionType>tag</scmVersionType>
                     <scmVersion>${pagopa-ecommerce-commons.version}</scmVersion>
+                    <scmVersionType>branch</scmVersionType>
+                    <scmVersion>CHK-2257-apm-auth-request</scmVersion>
```
Comment (pietro-tota) [inline pom.xml:489]:
```suggestion
```
Remove after commons library merge

File: src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestAuthorizationHandlerTest.java [hunk 2]
```diff
@@ -2262,4 +2263,219 @@ void shouldThrowExceptionNoGatewayMatchesDuringAuthorizationForWallet() {
...
+                                rptId,
+                                amount,
+                                description,
+                                nullPaymentContextCode,
+                                new ArrayList<>(),
+                                false
+                        )
+                ), // TODO
+                   // TRANSFER
+                   // LIST
```
Comment (giovanniberti) [inline src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestAuthorizationHandlerTest.java:2409]: 👀

**Review by giovanniberti — COMMENTED** [review]
👏

File: src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestAuthorizationHandlerTest.java (hunk 2 above)
Comment (pietro-tota) [reply src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestAuthorizationHandlerTest.java]: Fixed with https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/385/commits/19e6e9722fa921fbaecf7908b643614a684c2598 removing misleading comment (copy-past from previous tests). For the scope of this test transfert list is not used

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionRequestAuthorizationHandlerCommon.java [hunk 3]
```diff
@@ -94,52 +95,143 @@ protected Mono<Tuple2<String, String>> vposAuthRequestPipeline(AuthorizationRequ
...
+                }
+                ).map(
+                        orderIdAndFieldsDto -> Tuples.of(
+                                orderIdAndFieldsDto.getT1(),
+                                orderIdAndFieldsDto.getT2().getUrl(), // safe here, return url is checked in above
+                                // filter
+                                Optional.empty(),
+                                Optional.of(orderIdAndFieldsDto.getT2().getSessionId())
+                        )
+                );// safe here, sessionId is checked in requestNpgBuildSession method
```
[the remark this answers is not in the document]
Comment (pietro-tota) [reply src/main/java/it/pagopa/transactions/commands/handlers/TransactionRequestAuthorizationHandlerCommon.java]: Add an explanatory comment with [this commit](https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/385/commits/98d6afcb67b6b4c38b8007a6024e2b5a642a9f16).

**Review by giovanniberti — APPROVED** [review]

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**